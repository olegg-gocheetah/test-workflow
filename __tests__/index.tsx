import { render } from "@testing-library/react";
import Home from "../pages/index";
console.log(Home);

jest.mock("next/head", () => ({
  __esModule: true,
  default: ({ children }: any) => <span data-component="Head">{children}</span>,
}));

jest.mock("next/image", () => ({
  __esModule: true,
  default: (props: any) => <span data-component="Image" {...props} />,
}));

describe("Hello World!", () => {
  it("should touch home component", () => {
    const sut = render(<Home />);

    expect(sut.asFragment()).toMatchSnapshot();
  });
});
