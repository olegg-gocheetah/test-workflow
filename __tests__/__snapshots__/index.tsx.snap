// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World! should touch home component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <span
      data-component="Head"
    >
      <title>
        Create Next App
      </title>
      <meta
        content="Generated by create next app"
        name="description"
      />
      <link
        href="/favicon.ico"
        rel="icon"
      />
    </span>
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="description"
      >
        Get started by editing 
        <code
          class="code"
        >
          pages/index.tsx
        </code>
      </p>
      <div
        class="grid"
      >
        <a
          class="card"
          href="https://nextjs.org/docs"
        >
          <h2>
            Documentation →
          </h2>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="card"
          href="https://nextjs.org/learn"
        >
          <h2>
            Learn →
          </h2>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="card"
          href="https://github.com/vercel/next.js/tree/canary/examples"
        >
          <h2>
            Examples →
          </h2>
          <p>
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="card"
          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2>
            Deploy →
          </h2>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <span
          class="logo"
        >
          <span
            alt="Vercel Logo"
            data-component="Image"
            height="16"
            src="/vercel.svg"
            width="72"
          />
        </span>
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
